### ISSUE TITLE
Mac OS X Support

### ISSUE THREAD

**Issue by matthewelse** [matthewelse/bleep#1]
See notes for possible ways of doing this: e.g. PyObjC or perhaps the way that noble does it.

This may also require some refactoring, so that we can have different backends which can then be selected.

**Comment by diimdeep** [comment]
I tried building under OSX and get following:

```
src/XpcConnectionBase.cpp:72:27: warning: comparison of integers of different signs: 'int' and 'size_type' (aka 'unsigned long') [-Wsign-compare]
        for (int i = 0; i < str.size(); i++) {
                        ~ ^ ~~~~~~~~~~
1 warning generated.
ld: library not found for -lboost_python-py34
clang: error: linker command failed with exit code 1 (use -v to see invocation)
error: Setup script exited with error: command 'clang++' failed with exit status 1
```

```
sw_vers -productVersion
10.11.6
python --version
Python 3.5.2
```

**Comment by dsully** [comment]
You need to use Python 2.7 and "brew install boost-python"  or build Boost manually with Python 3.5+ support.

**Comment by silbo** [comment]
@matthewelse There is a typo in the README. It's `boost-python` not `python-boost`. Thanks!

**Comment by matthewelse** [comment]
@silps fixed in 3f252a3
